apollo-ios/Sources/Apollo/DefaultInterceptorProvider.swift

```diff
@@ -38,12 +38,21 @@ open class DefaultInterceptorProvider: InterceptorProvider {
         NetworkFetchInterceptor(client: self.client),
         ResponseCodeInterceptor(),
         MultipartResponseParsingInterceptor(),
-        JSONResponseParsingInterceptor(),
+        jsonParsingInterceptor(for: operation),
         AutomaticPersistedQueryInterceptor(),
         CacheWriteInterceptor(store: self.store),
     ]
   }
 
+  private func jsonParsingInterceptor<Operation: GraphQLOperation>(for operation: Operation) -> any ApolloInterceptor {
+    if (Operation.deferredFragments?.isEmpty ?? true) {
```

Comment:
We are calling `deferredFragments.isEmpty` in mutliple places. Maybe create a computed property that is more clear on the intention here.

Reply:
Done in https://github.com/apollographql/apollo-ios-dev/pull/217/commits/12ca32c502448d358d9d4a8a1022996e28e7e989.